core/src/main/java/hudson/node_monitors/InodesMonitor.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, Baptiste Mathus
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.node_monitors;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.text.ParseException;
+import java.util.logging.Logger;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import hudson.Extension;
+import hudson.model.Computer;
+import hudson.node_monitors.AbstractAsyncNodeMonitorDescriptor;
+import hudson.node_monitors.NodeMonitor;
+import hudson.remoting.Callable;
+import hudson.slaves.OfflineCause;
+import jenkins.model.Jenkins;
+import jenkins.security.MasterToSlaveCallable;
+
+/**
+ * Checks the used percentage of inodes on the FS (Linux/Unix only).
+ *
+ * Uses the <code>df</code> system command.
+ */
+public class InodesMonitor extends NodeMonitor {
+
+	private static final Logger LOGGER = Logger.getLogger(InodesMonitor.class.getSimpleName());
+
+	private static final String DEFAULT_OFFLINE_THRESHOLD = "95%";
```

Comment:
At first, I though about using the number value, and then I realized this may be simpler to use percentage, as I also wanted to enable the auto-offline feature too. And as my current code doesn't support commas, this may indeed be an issue with big configs.